File: output/ChatGPT/codenet/Java/C++/s047401835.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> numbers(N);

    for (int i = 0; i < N; i++) {
        cin >> numbers[i];
    }

    int count = 0;

    do {
        for (int i = 0; i < N; i++) {
            if (numbers[i] % 2 == 0) {
                numbers[i] /= 2;
            }
        }
        if (numbers.size() == N) count++;
    } while(numbers.size() == N);

    cout << count << endl;

    return 0;
}

File: output/StarCoder/codenet/Java/Python/s047401835.py
import sys

N = int(input())
numbers = []

for i in range(N):
    numbers.append(int(input()))

count = 0

while True:
    numbers = list(filter(lambda x: x % 2 == 0, numbers))
    numbers = list(map(lambda x: x / 2, numbers))
    if len(numbers) == N:
        count += 1
    else:
        break

print(count)